### ISSUE TITLE
Nested elements with same name as parent fails

### ISSUE THREAD

**Issue by mattroberts297** [databricks/spark-xml#160]
This XML:

```scala
<?xml version="1.0" encoding="UTF-8"?>
<body><p><p>Nested element</p><b>Important element</b></p></body>
```

When loaded using the XmlInputFormat (I think this underlies everything but may be wrong):

```scala
val records = sc.newAPIHadoopFile[LongWritable, Text, XmlInputFormat](path)
records.values.map(text => text.toString).foreach(log.info)
```

Logs this:

```bash
INFO Job$: <p><p>Nested element</p>
```

Instead of this:

```bash
<p><p>Nested element</p><b>Important element</b></p>
```

My workaround, use an xslt to rename the nested `<p>`:

```scala
<?xml version="1.0" encoding="UTF-8"?>
<body><p><a>Nested element</a><b>Important element</b></p>
</body>
```

I think the solution is to make `XmlRecordReader` track the depth (incrementing on start tags and decrementing on end tags) and only stop reading when the end tag is encountered and the depth is 0. Not entirely trivial and I think the bug exists in Mahout's Java implementation of [`XmlRecordReader`](https://github.com/apache/mahout/blob/ad84344e4055b1e6adff5779339a33fa29e1265d/examples/src/main/java/org/apache/mahout/classifier/bayes/XmlInputFormat.java) too. If you agree it's a bug I'll try and raise a PR for this.

On an aside, with the xslt'd version of my file the library works really well. Thanks!

**Comment by HyukjinKwon** [comment]
Oh, yea. Thanks for detailed explanation. Yes, this is a bug and we should fix. Actually, this is a duplicate of https://github.com/databricks/spark-xml/issues/148 :)

Please go ahead for a PR. 

BTW, please make sure your commits are linked to you. I just merged your PR but realised that the commits are not linked to your.

**Comment by HyukjinKwon** [comment]
Let me mark this as a duplicate. Let's close both if this one is solved (as this one has a good example and explanation)

**Comment by mattroberts297** [comment]
> BTW, please make sure your commits are linked to you. I just merged your PR but realised that the commits are not linked to your github account.

Thanks for the heads up (bad config on my part).

**Comment by HyukjinKwon** [comment]
This is fixed in #161
